js/index.js

```diff
@@ -1,93 +1,141 @@
+        
+        console.log(username, email, message);
+
+        const messageSection = document.querySelector('#messages');
+        const messageList = messageSection.querySelector('ul');
+        const newMessage = document.createElement('li'); //makes a new list item
+        messageList.appendChild(newMessage);
+
+        newMessage.innerHTML = `
+        <a href="mailto:${email}">${username}</a>
+        <span> ${message}</span>`;
+
+        // Remove button creation
+        const removeButton = document.createElement('button');
+        removeButton.innerHTML = "Remove";
+        removeButton.setAttribute("type", "button");
+        removeButton.setAttribute("class", "remove_button");
+
+        removeButton.addEventListener('click', function(){
+            const entry = removeButton.parentNode;
+            entry.removeChild(removeButton);
+            newMessage.remove();
+
+        // To hide the message section when empty
+            if(messageList.children.length === 0) {
+                messageSection.style.display = 'none';
+                leaveMessage.append(footer);
+            }
+        });  
     
-    const username = event.target.usersName.value;
-    const email = event.target.usersEmail.value;
-    const message = event.target.usersMessage.value;
-    
-    console.log(username, email, message);
-
-    const messageSection = document.querySelector('#messages');
-    const messageList = messageSection.querySelector('ul');
-    const newMessage = document.createElement('li'); //makes a new list item
-    messageList.appendChild(newMessage);
-
-    newMessage.innerHTML = `
-    <a href="mailto:${email}">${username}</a>
-    <span> ${message}</span>`;
-
-    // Remove button creation
-    const removeButton = document.createElement('button');
-    removeButton.innerHTML = "Remove";
-    removeButton.setAttribute("type", "button");
-    removeButton.setAttribute("class", "remove_button");
-
-    removeButton.addEventListener('click', function(){
-        const entry = removeButton.parentNode;
-        entry.removeChild(removeButton);
-        newMessage.remove();
-
-    // To hide the message section when empty
-        if(messageList.children.length === 0) {
-            messageSection.style.display = 'none';
-            leaveMessage.append(el);
+    //   Creating the Edit button
+        const editButton = document.createElement('button');
+        editButton.innerHTML = "Edit";
+        editButton.setAttribute("type", "button");
+        editButton.setAttribute("class", "edit_button");
+
+        editButton.addEventListener('click', function(){
+        newMessage.innerHTML = `<a href="mailto:${email}">${username}</a>
+        <textarea> ${message}</textarea>`
+
+        //Creating the second Submit button
+        const resubmitButton = document.createElement('button');
+        resubmitButton.innerHTML = "Submit";
+        resubmitButton.setAttribute("type", "submit");
```

Comment:
To prevent the page from reloading, you could change the button type to just `button` and handle everything in a `click` event listener.

Reply:
When I change the type of resubmit button to 'button', the event is triggered when I click on 'Edit' button